src/WorkerChannel.ts

```diff
@@ -61,4 +66,11 @@ export class WorkerChannel {
             after(context);
         }
     }
+
+    public async initAppDir(dir: string) {
+        if (dir !== this.functionAppDir) {

```

Comment:
In the case of `functionEnvironmentReloadRequest`, I feel like we would want to re-read the package.json file regardless of if the dir changed, just in case the file contents changed

Reply:
Done